src/topology/separation.lean

```diff
@@ -16,6 +16,59 @@ variables {α : Type u} {β : Type v} [topological_space α]
 
 section separation
 
+/--
+`separate` is a predicate on pairs of `finset`s of a topological space.  It holds if the two
+`finset`s are contained in disjoint open sets.
+-/
+def separate : finset α → finset α → Prop :=
+  λ (s t : finset α), ∃ U V : (set α), (is_open U) ∧ is_open V ∧
+  (∀ a : α, a ∈ s → a ∈ U) ∧ (∀ a : α, a ∈ t → a ∈ V) ∧ disjoint U V
+
+lemma separate.symm {s t : finset α} : separate s t → separate t s :=
+begin
+  rintros ⟨U, V, oU, oV, aU, bV, UV⟩,
+  exact ⟨V, U, oV, oU, bV, aU, disjoint.symm UV⟩
+end
+
+lemma separate.comm (s t : finset α) : separate s t ↔ separate t s :=
+⟨separate.symm, separate.symm⟩
+
+lemma separate.empty_right : ∀ {a : finset α}, separate a ∅ :=
+λ a, ⟨_, _, is_open_univ, is_open_empty, λ a h, mem_univ a, λ a h, by cases h, disjoint_empty _⟩
+
+lemma separate.empty_left : ∀ {a : finset α}, separate ∅ a :=
+λ _, separate.symm separate.empty_right
+
+lemma separate.union_left :
+  ∀ {a b c : finset α}, separate a c → separate b c → separate (a ∪ b) c :=
+begin
+  rintros a b c ⟨U, V, oU, oV, aU, bV, UV⟩ ⟨W, X, oW, oX, aW, bX, WX⟩,
+  refine ⟨U ∪ W, V ∩ X, is_open_union oU oW, is_open_inter oV oX,
+    λ x xab, _, λ x xc, ⟨bV _ xc, bX _ xc⟩, _⟩,
+  { cases finset.mem_union.mp xab with h h,
+    { exact mem_union_left W (aU x h) },
+    { exact mem_union_right U (aW x h) } },
+  { apply set.disjoint_union_left.mpr,
+    exact ⟨disjoint_of_subset_right (inter_subset_left _ _) UV,
+      disjoint_of_subset_right (inter_subset_right _ _) WX⟩ },
+end
+
+lemma separate.union_right :
```

Comment:
Again, you can put five arguments left of the comma.
Also, if you want to practice the projection notation, you can write `(ab.symm.union_left ac.symm).symm` (if I didn't mess up)